engine/class_modules/warlock/sc_warlock.cpp

```diff
@@ -55,6 +55,66 @@ struct drain_life_t : public warlock_spell_t
   }
 };  
 
+struct impending_catastrophe_dot_t : public warlock_spell_t
+{
+  impending_catastrophe_dot_t( warlock_t* p )
+    : warlock_spell_t( "impending_catastrophe_dot", p, p->find_spell( 322170 ) )
+  {
+    background = true;
+    may_miss   = false;
+    dual       = true;
+    aoe        = -1;
+    dot_duration *= 1 + p->conduit.catastrophic_origin.percent();
```

Comment:
Dunno if it works like that, but the conduit reads
  `Impending Catastrophe's damage over time and the inflicted Curse last |cFFFFFFFF${$s1}.1%|r longer against your primary target.`
If it actually does work like that, you'll have to change this so the dot duration isn't extended for all targets hit.

Pretty sure what you could do is to override `action_t::composite_dot_duration()` with smth like
```cpp
timespan_t composite_dot_duration( const action_state_t* s ) const override
{
  if ( s->chain_target = 0 ) // main target
    return dot_duration * ( 1 + p->conduit.catastrophic_origin.percent() );
  return dot_duration;
}
```